connectors/http/polling/src/main/java/io/camunda/connector/http/polling/task/PollingOperateTask.java

```diff
@@ -0,0 +1,99 @@
+
+import io.camunda.connector.api.inbound.InboundIntermediateConnectorContext;
+import io.camunda.connector.api.inbound.ProcessInstanceContext;
+import io.camunda.connector.http.base.services.HttpService;
+import io.camunda.connector.http.polling.service.HttpPollingSharedExecutorService;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * PollingOperateTask is responsible for handling HTTP polling operations. Each instance of this
+ * class is used for one flowNode and manages polling across all its corresponding processInstances.
+ */
+public class PollingOperateTask extends AbstractScheduledTask {
+
+  private final InboundIntermediateConnectorContext context;
+  private final HttpService httpService;
+  private final HttpPollingSharedExecutorService executorService;
+  private final Duration httpRequestInterval;
+  private final Map<String, HttpRequestTask> requestTaskMap;
+
+  public PollingOperateTask(
+      InboundIntermediateConnectorContext context,
+      HttpService httpService,
+      HttpPollingSharedExecutorService executorService,
+      Duration httpRequestInterval) {
+    this.context = context;
+    this.httpService = httpService;
+    this.executorService = executorService;
+    this.httpRequestInterval = httpRequestInterval;
+    this.requestTaskMap = new ConcurrentHashMap<>();
+  }
+
+  @Override
+  public void run() {
+    List<ProcessInstanceContext> processInstanceContexts = context.getProcessInstanceContexts();
+    if (processInstanceContexts != null) {
+      removeInactiveTasks(processInstanceContexts);
+      scheduleOrRefreshTasks(processInstanceContexts);
+    }
+  }
+
+  private void removeInactiveTasks(final List<ProcessInstanceContext> processInstanceContexts) {
+    List<String> activeTasks =
+        processInstanceContexts.stream()
+            .map(ProcessInstanceContext::getKey)
+            .map(key -> context.getDefinition().elementId() + key)
+            .toList();
+
+    List<String> inactiveTasks =
+        requestTaskMap.keySet().stream().filter(key -> !activeTasks.contains(key)).toList();
+
+    inactiveTasks.forEach(
+        inactiveTaskKey -> {
+          executorService.cancelTask(inactiveTaskKey);
+          requestTaskMap.remove(inactiveTaskKey);
+        });
+  }
+
+  /**
+   * Schedules or refreshes HttpRequestTasks based on the given processInstanceContexts. Each
+   * ProcessInstanceContext and flow node together uniquely define a single HttpRequestTask.
+   */
+  private void scheduleOrRefreshTasks(List<ProcessInstanceContext> contexts) {
+    contexts.forEach(
+        processInstanceContext -> {
+          String taskKey = context.getDefinition().elementId() + processInstanceContext.getKey();
+          // Check if an HttpRequestTask already exists for this unique task key.
+          if (executorService.containsTask(taskKey)) {
```

Comment:
Why are there two places? Lets just have one place where tasks are tracked. Can just be this class.

Reply:
Pleae look at my other comment below. We agreed on that its provided by the executor service.